projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/answers/ParseAnswerElement.java

```diff
@@ -0,0 +1,25 @@
+package org.batfish.datamodel.answers;
+
+import java.util.SortedMap;
+import org.batfish.common.BatfishException;
+import org.batfish.common.ParseTreeSentences;
+import org.batfish.common.Warnings;
+
+public interface ParseAnswerElement extends AnswerElement {
+
+  SortedMap<String, BatfishException.BatfishStackTrace> getErrors();
+
+  SortedMap<String, ParseStatus> getParseStatus();
+
+  SortedMap<String, ParseTreeSentences> getParseTrees();
+
+  SortedMap<String, Warnings> getWarnings();
+
+  void setErrors(SortedMap<String, BatfishException.BatfishStackTrace> errors);
+
+  void setParseStatus(SortedMap<String, ParseStatus> parseStatus);
+
+  void setParseTrees(SortedMap<String, ParseTreeSentences> parseTrees);
+
+  void setWarnings(SortedMap<String, Warnings> warnings);
```

Comment:
dumb q: should all answer elements have optional errors and warnings, not just parsing-related?

Reply:
That would require a huge structural change. It might not be a bad idea, though. Definitely out of scope for this PR IMO.